src/lib/core/PeerId.h

```diff
@@ -17,51 +17,54 @@
 
 #pragma once
 
+#include <lib/core/DataModelTypes.h>
 #include <lib/core/NodeId.h>
 
 namespace chip {
 
-using CompressedFabricId = uint64_t;
 using FabricId           = uint64_t;
 
-constexpr CompressedFabricId kUndefinedCompressedFabricId = 0ULL;
-
 constexpr FabricId kUndefinedFabricId = 0ULL;
 constexpr uint16_t kUndefinedVendorId = 0U;
 
-/// A peer is identified by a node id within a compressed fabric ID
+/// A peer is identified by a node id within a fabric.
 class PeerId
 {
 public:
-    PeerId() {}
+    PeerId() : mNodeId(kUndefinedNodeId), mFabricIndex(kUndefinedFabricIndex) {}
+    PeerId(NodeId nodeId, FabricIndex fabricIndex) : mNodeId(nodeId), mFabricIndex(fabricIndex) {}
+
+    PeerId(const PeerId &) = default;
+    PeerId(PeerId &&) = default;
+    PeerId & operator=(const PeerId &) = default;
+    PeerId & operator=(PeerId &&) = default;
 
     NodeId GetNodeId() const { return mNodeId; }
-    PeerId & SetNodeId(NodeId id)
-    {
-        mNodeId = id;
-        return *this;
-    }
+    FabricIndex GetFabricIndex() const { return mFabricIndex; }
 
-    CompressedFabricId GetCompressedFabricId() const { return mCompressedFabricId; }
-    PeerId & SetCompressedFabricId(CompressedFabricId id)
-    {
-        mCompressedFabricId = id;
-        return *this;
+    bool operator==(const PeerId & other) const {
+        if (mFabricIndex == kUndefinedFabricIndex)
```

Comment:
So this `operator==` is not symmetric, right?  If `this` has `mFabricIndex == kUndefinedFabricIndex` but `other` does not, then we could have `*this == other` but `other != *this`.

Is this purposeful?  It's not a great behavior for an `operator==`.